LibRangeCheck-2.0/LibRangeCheck-2.0.lua

```diff
@@ -511,17 +583,18 @@ local function findSpellIdx(spellName)
         return nil
     end
     for i = 1, getNumSpells() do
-        local spell, rank = GetSpellBookItemName(i, BOOKTYPE_SPELL)
-        if spell == spellName then return i end
+        local spell = GetSpellBookItemName(i, BOOKTYPE_SPELL)
+        if spell == spellName then
+            return i
+        end
     end
     return nil
 end
 
 -- minRange should be nil if there's no minRange, not 0
 local function addChecker(t, range, minRange, checker, info)
     local rc = { ["range"] = range, ["minRange"] = minRange, ["checker"] = checker, ["info"] = info }
-    for i = 1, #t do
-        local v = t[i]
+    for i, v in next, t do
```

Comment:
All these conversions of 1, #table to next, t are strange. They switch from in index-order iteration to unordered iteration. No idea whether that has any relevance, but it's at least worse performance.

Reply:
No, see https://www.lua.org/pil/7.3.html

A indexed iteration is essentially a ipairs iteration, which is not the same as using **next**